platform/ext/target/nordic_nrf/common/core/cmsis_drivers/Driver_USART.c

```diff
@@ -22,6 +22,11 @@
 #include <nrfx_uarte.h>
 #include <string.h>
 
+#if !(DOMAIN_NS == 1U) && defined(SECURE_UART1) && defined(CONFIG_TFM_LOG_SHARE_UART)
```

Comment:
Let's put a config check in CMake on CONFIG_TFM_LOG_SHARE_UART requiring SECURE_UART1.

Then we won't accidentally have a config where the UART is supposedly shared, but no logging comes out.

This expression is also simplified.